<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Forensic likelihood ratios for whole-genome sequencing accounting for errors.">
<title>Whole-genome sequencing genotype error estimation and forensic likelihood ratios accounting for errors • wgsLR</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Whole-genome sequencing genotype error estimation and forensic likelihood ratios accounting for errors">
<meta property="og:description" content="Forensic likelihood ratios for whole-genome sequencing accounting for errors.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">wgsLR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1.9010</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/introduction.html">Introduction</a>
    <a class="dropdown-item" href="articles/using-stan.html">Using Stan for Bayesian inference</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/mikldk/wgsLR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="wgslr-shotgun-sequencing-for-human-identification-dynamic-snp-marker-sets-and-likelihood-ratio-calculations-accounting-for-errors">
<code>wgsLR</code>: Shotgun sequencing for human identification: Dynamic SNP marker sets and likelihood ratio calculations accounting for errors<a class="anchor" aria-label="anchor" href="#wgslr-shotgun-sequencing-for-human-identification-dynamic-snp-marker-sets-and-likelihood-ratio-calculations-accounting-for-errors"></a>
</h1></div>
<p>Please refer to the online documentation at <a href="https://mikldk.github.io/wgsLR" class="external-link uri">https://mikldk.github.io/wgsLR</a>, including the vignettes.</p>
<div class="section level2">
<h2 id="scientific-publication">Scientific publication<a class="anchor" aria-label="anchor" href="#scientific-publication"></a>
</h2>
<p>The research associated with this software is described in</p>
<blockquote>
<p>Andersen, M. M., Kampmann, M.-L., Jepsen, A. H., Morling, N., Eriksen, P. S., Børsting, C., &amp; Andersen, J. D. (2025). <em>Shotgun DNA sequencing for human identification: Dynamic SNP selection and likelihood ratio calculations accounting for errors.</em> Forensic Science International: Genetics, 74, 103146. <a href="https://doi.org/10.1016/j.fsigen.2024.103146" class="external-link">doi:10.1016/j.fsigen.2024.103146</a>.</p>
</blockquote>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install from Github with vignettes run this command from within <code>R</code> (please install <code>remotes</code> first if not already installed):</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># install.packages('remotes')</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"mikldk/wgsLR"</span>, <span class="at">build_vignettes =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>You can also install the package without vignettes if needed as follows:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"mikldk/wgsLR"</span>)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="a-few-small-examples">A few small examples<a class="anchor" aria-label="anchor" href="#a-few-small-examples"></a>
</h2>
<div class="section level3">
<h3 id="estimating-the-genotype-error-probability-w">Estimating the genotype error probability, <span class="math inline"><em>w</em></span>
<a class="anchor" aria-label="anchor" href="#estimating-the-genotype-error-probability-w"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cases</span> <span class="op">&lt;-</span> <span class="fu">wgsLR</span><span class="fu">::</span><span class="fu"><a href="reference/sample_data_Hp_w.html">sample_data_Hp_w</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">1000</span>, w <span class="op">=</span> <span class="fl">0.1</span>, p <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.25</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">cases</span><span class="op">$</span><span class="va">X_D</span>, <span class="va">cases</span><span class="op">$</span><span class="va">X_S</span><span class="op">)</span></span>
<span><span class="va">tab</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="do">##    </span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="do">##       0   1   2</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="do">##   0 180  55   9</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="do">##   1  59 188  88</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="do">##   2   4  89 328</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">w_mle</span> <span class="op">&lt;-</span> <span class="fu">wgsLR</span><span class="fu">::</span><span class="fu"><a href="reference/estimate_w.html">estimate_w</a></span><span class="op">(</span><span class="va">tab</span><span class="op">)</span></span>
<span><span class="va">w_mle</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="do">## [1] 0.09596158</span></span></code></pre></div>
<div class="section level4">
<h4 id="cautionary-note-not-just-standard-vcf-files">Cautionary note: not just standard VCF files<a class="anchor" aria-label="anchor" href="#cautionary-note-not-just-standard-vcf-files"></a>
</h4>
<p>It is necessary to obtain sequencing results for all bases in the selected segments (including read depth and genotype quality). Thus, it is <strong>not</strong> sufficient to just use information from “confirmed”/high probability variants from the reference genome (variants identified in standard vcf-file format), as this can introduce bias in the results. <strong>Information from all bases in the chosen genomic areas of interest is needed</strong>. One way to achieve this by using <a href="https://gatk.broadinstitute.org/hc/en-us/articles/9570334998171-HaplotypeCaller" class="external-link">GATK HaplotypeCaller</a> with the additional argument <code>--emit-ref-confidence BP_RESOLUTION</code> for the genomic areas of interest (using <code>-L areas.interval_list</code>).</p>
</div>
</div>
<div class="section level3">
<h3 id="calculating-likelihood-ratios-lrs">Calculating likelihood ratios (<span class="math inline"><em>L</em><em>R</em></span>’s)<a class="anchor" aria-label="anchor" href="#calculating-likelihood-ratios-lrs"></a>
</h3>
<p>Assume that a trace sample had four loci with genotypes (0/1 = 1, 0/0 = 0, 1/1 = 2, 1/1 = 2). A person of interest is then typed for the same four loci and has genotypes (0/0 = 0, 0/0 = 0, 1/1 = 2, 1/1 = 2), i.e. a mismatch on the first locus.</p>
<p>For simplicity, assume that the genotype probabilites are P(0/0 = 0) = 0.25, P(0/1 = 1/0 = 1) = 0.25, and P(1/1 = 2) = 0.5.</p>
<p>If no errors are possible, then <span class="math inline"><em>w</em> = 0</span> and</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">wgsLR</span><span class="fu">::</span><span class="fu"><a href="reference/calc_LRs_w.html">calc_LRs_w</a></span><span class="op">(</span>xs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span>, </span>
<span>                  xd <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span>, </span>
<span>                  w <span class="op">=</span> <span class="fl">0</span>, </span>
<span>                  p <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.25</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="do">## [1] 0 4 2 2</span></span></code></pre></div>
<p>and the product is 0 due to the mismatch at the first locus.</p>
<p>If instead we acknowledge that errors are possible, then for <span class="math inline"><em>w</em> = 0.001</span> we obtain that</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">LR_contribs</span> <span class="op">&lt;-</span> <span class="fu">wgsLR</span><span class="fu">::</span><span class="fu"><a href="reference/calc_LRs_w.html">calc_LRs_w</a></span><span class="op">(</span>xs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span>, </span>
<span>                                 xd <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span>, </span>
<span>                                 w <span class="op">=</span> <span class="fl">0.001</span>, </span>
<span>                                 p <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.25</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">LR_contribs</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="do">## [1] 0.01192834 3.99199202 1.99799850 1.99799850</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/prod.html" class="external-link">prod</a></span><span class="op">(</span><span class="va">LR_contribs</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="do">## [1] 0.1900904</span></span></code></pre></div>
<p>We can also consider the <span class="math inline"><em>L</em><em>R</em></span>s for a range for plausible values of <span class="math inline"><em>w</em></span>:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ws</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1e-6</span>, <span class="fl">1e-3</span>, <span class="fl">1e-2</span>, <span class="fl">1e-1</span><span class="op">)</span></span>
<span><span class="va">LRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">ws</span>, \<span class="op">(</span><span class="va">w</span><span class="op">)</span> <span class="fu">wgsLR</span><span class="fu">::</span><span class="fu"><a href="reference/calc_LRs_w.html">calc_LRs_w</a></span><span class="op">(</span>xs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span>, </span>
<span>                                         xd <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span>, </span>
<span>                                         w <span class="op">=</span> <span class="va">w</span>, </span>
<span>                                         p <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.25</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>                <span class="fu"><a href="https://rdrr.io/r/base/prod.html" class="external-link">prod</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>log10w <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">ws</span><span class="op">)</span>, w <span class="op">=</span> <span class="va">ws</span>, </span>
<span>           LR <span class="op">=</span> <span class="va">LRs</span>, WoElog10LR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">LRs</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="do">##   log10w     w           LR WoElog10LR</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a><span class="do">## 1     -6 1e-06 0.0001919981 -3.7167031</span></span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a><span class="do">## 2     -3 1e-03 0.1900903523 -0.7210399</span></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a><span class="do">## 3     -2 1e-02 1.7379421372  0.2400353</span></span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a><span class="do">## 4     -1 1e-01 7.1409934157  0.8537586</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="different-error-rates">Different error rates<a class="anchor" aria-label="anchor" href="#different-error-rates"></a>
</h2>
<p>Assume that the trace donor profile has <span class="math inline"><em>w</em><sub><em>D</em></sub> = 10<sup>−4</sup></span> and the suspect reference profile has <span class="math inline"><em>w</em><sub><em>S</em></sub> = 10<sup>−8</sup></span>. Then the <span class="math inline"><em>L</em><em>R</em></span> is:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">LR_contribs</span> <span class="op">&lt;-</span> <span class="fu">wgsLR</span><span class="fu">::</span><span class="fu"><a href="reference/calc_LRs_wDwS.html">calc_LRs_wDwS</a></span><span class="op">(</span>xs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span>, </span>
<span>                                    xd <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span>, </span>
<span>                                    wD <span class="op">=</span> <span class="fl">1e-4</span>, </span>
<span>                                    wS <span class="op">=</span> <span class="fl">1e-8</span>,</span>
<span>                                    p <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.25</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/prod.html" class="external-link">prod</a></span><span class="op">(</span><span class="va">LR_contribs</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="do">## [1] 0.01279168</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/mikldk/wgsLR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/mikldk/wgsLR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>GPL</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing wgsLR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Mikkel Meyer Andersen <br><small class="roles"> Author, maintainer, copyright holder </small>  </li>
<li>Poul Svante Eriksen <br><small class="roles"> Author, copyright holder </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Mikkel Meyer Andersen, Poul Svante Eriksen.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
